common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -389,6 +407,7 @@ protected boolean runAllTasks(long timeoutNanos) {
         fetchFromScheduledTaskQueue();
         Runnable task = pollTask();
         if (task == null) {
+            afterRunningAllTasks();
```

Comment:
So to run it in the case this method throws an exception?
I don't see that we have such guarantees here, eg: Whether the remaining tasks be executed if this code throws after executing `n` tasks? Also, it would mean that we run the tail tasks after fatal errors (OOM eg). 
If this was cleanup code, I would do it in a finally block, but I am open to be convinced!